pages/selfCheckQuestions.tsx

```diff
@@ -0,0 +1,284 @@
+        options: ["1", "2", "3"],
+        alphanumericInput: true,
+        questionNumber: 3,
+    },
+    {
+        _id: "60e642d7e4a1ae34207a92a6",
+        type: "long_answer",
+        question: "autem sunt eiusdolores nesciunt impedit?",
+        options: ["1", "2", "3"],
+        alphanumericInput: true,
+        questionNumber: 4,
+    },
+    {
+        _id: "60e642d7e4a1ae34207a92a7",
+        type: "slider",
+        question: "autem sunt eiusdolores nesciunt impedit?",
+        options: ["1", "2", "3", "4", "5"],
+        alphanumericInput: true,
+        questionNumber: 5,
+    },
+];
+
+//Self Check Questions React functional component
+const Home: React.FC = () => {
+    const [list, setList] = React.useState(questionList);
+    const [count, setCount] = React.useState(questionList.length + 1);
+    const newSliderRange0 = [];
+    for (let i = 1; i <= 3; i++) {
+        newSliderRange0.push(i);
+    }
+    const [sliderRange, setSliderRange] = React.useState(newSliderRange0);
+    function changeQuestionType(id, target) {
+        const newList = list.slice(0);
+        newList[newList.findIndex((e) => e._id === id)].type = target;
+
+        setList(newList);
+    }
+    function changeAlphanumeric(id, target) {
+        const newList = list.slice(0);
+        newList[newList.findIndex((e) => e._id === id)].alphanumericInput =
+            target;
+        setList(newList);
+    }
+    function addOneOption(id) {
+        const newList = list.slice(0);
+        newList[newList.findIndex((e) => e._id === id)].options = [
+            ...newList[newList.findIndex((e) => e._id === id)].options,
+            `${
+                newList[newList.findIndex((e) => e._id === id)].options.length +
+                1
+            }`,
+        ];
+        setList(newList);
+    }
+
+    function removeOneOption(id, index) {
+        const newList = list.slice(0);
+        newList[newList.findIndex((e) => e._id === id)].options.splice(
+            index,
+            1,
+        );
+        setList(newList);
+    }
+    function changeSliderOption(id, lowerBound, upperBound) {
+        const optionList = [];
+        const newList = list.slice(0);
+        lowerBound = parseInt(lowerBound);
+        upperBound = parseInt(upperBound);
+        const newSliderRange = [];
+
+        for (let i = lowerBound; i <= upperBound; i++) {
+            optionList.push(String(i));
+            newSliderRange.push(i);
+        }
+
+        newList[newList.findIndex((e) => e._id === id)].options = optionList;
+        setSliderRange(newSliderRange);
+        setList(newList);
+    }
+    function moveQuesitonDown(index) {
+        const newList = list.slice(0);
+        const tempQuestion = newList[index];
+        newList[index] = newList[index + 1];
+        newList[index + 1] = tempQuestion;
+        setList(newList);
+    }
+    function moveQuesitonUp(index) {
+        const newList = list.slice(0);
```

Comment:
```suggestion
        newList[index] = newList.splice(index - 1, 1, newList[index])[0]
```
fancier way of doing the swap but you can leave this as is if it's easier to read